src/compiler/checker.ts

```diff
@@ -12691,8 +12691,11 @@ namespace ts {
                 }
             }
 
+            const isVariableInForInStatement = node.parent.parent.kind === SyntaxKind.ForInStatement;
             // For a binding pattern, check contained binding elements
-            if (isBindingPattern(node.name)) {
+            // NOTE: do not check them if this is a binding pattern in variable declaration of ForIn statement
+            // this situation is already illegal so no need to report misleading errors.
+            if (isBindingPattern(node.name) && !isVariableInForInStatement) {

```

Comment:
because we have not implemented this yet: #3715. I think I'll remove this check so it won't be missed once #3715 is addresssed